common/libs/wallet/non-deterministic/web3.ts

```diff
@@ -0,0 +1,92 @@
+import { IWallet } from '../IWallet';
+import { ExtendedRawTransaction } from 'libs/transaction';
+import { networkIdToName } from 'libs/values';
+import { addHexPrefix } from 'ethereumjs-util';
+
+export default class Web3Wallet implements IWallet {
+  private web3: any;
+  private address: string;
+  private network: string;
+
+  constructor(web3: any, address: string, network: string) {
+    this.web3 = web3;
+    this.address = address;
+    this.network = network;
+  }
+
+  public getAddressString(): Promise<string> {
+    return Promise.resolve(this.address);
+  }
+
+  public signRawTransaction(): Promise<string> {
+    return Promise.reject(
+      new Error('Web3 wallets cannot sign raw transactions.')
+    );
+  }
+
+  public signMessage(msg: string): Promise<string> {
+    return new Promise((resolve, reject) => {
+      const msgHex = Buffer.from(msg).toString('hex');
```

Comment:
you can use the method `bufferToHex` from ethereumjs-utils. It'll convert it to hex and add the prefix for you in one call